### PR TITLE
add msgpack-c

### PR DESCRIPTION
**Pull request by HookedBehemoth**
Depends on #167

### CODE REVIEW

File: switch/msgpack/switch-msgpack.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff -NBaur msgpack-3.3.0.switch/CMakeLists.txt msgpack-3.3.0/CMakeLists.txt
+--- msgpack-3.3.0/CMakeLists.txt	2018-12-02 13:20:52.000000000 +0000
++++ msgpack-3.3.0.switch/CMakeLists.txt	2018-12-23 18:49:07.178930622 +0000
+@@ -5,6 +5,8 @@
+     CMAKE_POLICY(SET CMP0054 NEW)
+ ENDIF ()
+ 
++set(CMAKE_C_COMPILER_WORKS TRUE)
```
Comment (WinterMute) [inline switch/msgpack/switch-msgpack.patch:8]: Setting these variables manually shouldn't be necessary, what error occurs without them?

**Comment by WinterMute** [comment]
The only dependency I can see on #167 is in the example code & the example building is disabled. Is there a reason you think there's still a dependency here?

File: switch/msgpack/switch-msgpack.patch (hunk 1 above)
Comment (HookedBehemoth) [reply switch/msgpack/switch-msgpack.patch:8]: When trying to compile the cmake test program it fails as _start isn't defined without linking against libnx

**Comment by HookedBehemoth** [comment]
Oh you are correct.
I closed the cJSON PR.

**Pull request closed without merging**